src/engine/TransitivePath.cpp

```diff
@@ -116,22 +113,27 @@ size_t TransitivePath::getResultWidth() const { return _resultWidth; }
 vector<ColumnIndex> TransitivePath::resultSortedOn() const {
   const std::vector<ColumnIndex>& subSortedOn =
       _subtree->getRootOperation()->getResultSortedOn();
-  if (_leftSideTree == nullptr && _rightSideTree == nullptr && _leftIsVar &&
-      _rightIsVar && subSortedOn.size() > 0 && subSortedOn[0] == _leftSubCol) {
+  if (_lhs.isBound() && _lhs.isVariable() && _rhs.isBound() &&
+      _lhs.isVariable() && subSortedOn.size() > 0 &&
+      subSortedOn[0] == _lhs.subCol) {
     // This operation preserves the order of the _leftCol of the subtree.
     return {0};
   }
-  if (_leftSideTree != nullptr) {
+  if (_lhs.treeAndCol.has_value()) {
+    auto tree = _lhs.treeAndCol.value().first;
+    auto col = _lhs.treeAndCol.value().second;
     const std::vector<ColumnIndex>& leftSortedOn =
-        _leftSideTree->getRootOperation()->getResultSortedOn();
-    if (leftSortedOn.size() > 0 && leftSortedOn[0] == _leftSideCol) {
+        tree->getRootOperation()->getResultSortedOn();
+    if (leftSortedOn.size() > 0 && leftSortedOn[0] == col) {
       return {0};
     }
   }
-  if (_rightSideTree != nullptr) {
+  if (_rhs.treeAndCol.has_value()) {
+    auto tree = _rhs.treeAndCol.value().first;
+    auto col = _rhs.treeAndCol.value().second;
     const std::vector<ColumnIndex>& rightSortedOn =
-        _rightSideTree->getRootOperation()->getResultSortedOn();
-    if (rightSortedOn.size() > 0 && rightSortedOn[0] == _rightSideCol) {
+        tree->getRootOperation()->getResultSortedOn();
+    if (rightSortedOn.size() > 0 && rightSortedOn[0] == col) {
```

Comment:
This can probably be delegated to a member function `TransitivePathSide::isSortedOnInputCol` or something similar.